### PR TITLE
add dns resources to service-level terraform

### PR DESCRIPTION
**Pull request by bobcallaway**
Signed-off-by: Bob Callaway <bcallaway@google.com>

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
Cool

File: terraform/gcp/modules/dex/variables.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+variable "project_id" {
+  type    = string
+  default = ""
+  validation {
+    condition     = length(var.project_id) > 0
+    error_message = "Must specify project_id variable."
+  }
+}
+
+variable "region" {
```
Comment (k4leung4) [inline terraform/gcp/modules/dex/variables.tf:26]: this doesnt seem to be used.
delete it?

File: terraform/gcp/modules/ctlog/variables.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+variable "project_id" {
+  type    = string
+  default = ""
+  validation {
+    condition     = length(var.project_id) > 0
+    error_message = "Must specify project_id variable."
+  }
+}
+
+variable "region" {
```
Comment (k4leung4) [inline terraform/gcp/modules/ctlog/variables.tf:26]: this doesnt seem to be used.
please delete

File: terraform/gcp/modules/ctlog/main.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  // Do not disable the service on destroy. On destroy, we are going to
+  // destroy the project, but we need the APIs available to destroy the
+  // underlying resources.
+  disable_on_destroy = false
+}
+
+resource "google_dns_record_set" "A_ctfe" {
+  name = "ctfe.${var.dns_domain_name}"
+  type = "A"
+  ttl  = 60
```
Comment (vaikas) [inline terraform/gcp/modules/ctlog/main.tf:34]: This seems awfully short to me, but my dns-fu is super old :)

**Review by vaikas — APPROVED** [review]
The 'pending' checks are because we changed the set of expected tests. All the tests were passing.

**Review by vaikas — APPROVED** [review]

**Pull request merged**